### PR TITLE
[cmake] partial installs

### PR DESCRIPTION
**Pull request by BurningEnlightenment**
Integrate the improvements from ned14/quickcpplib#23 and ned14/quickcpplib#24

### CODE REVIEW

**Comment by BurningEnlightenment** [comment]
@ned14 the installability action seems to checkout the current develop branch:
https://github.com/ned14/llfio/blob/8861d608734d52bb0a63b7fc5aae9fc17b36d634/.github/workflows/installability.yml#L41
... working as intended?

It also seems that the install componentization broke the current llfio master & develop installs in combination with the most recent quickcpplib commits due to the additional `ntkernel-error-category` target installs. This needs to be fixed by either merging this PR or reverting the commits on quickcpplib 😒

File: CMakeLists.txt [hunk 1]
```diff
@@ -475,14 +475,38 @@ endif()
 # Cache this library's auto scanned sources for later reuse
 include(QuickCppLibCacheLibrarySources)
 
+# Dependencies needed to consume our cmake package
+set(PROJECT_PACKAGE_DEPENDENCIES [=[
+include(CMakeFindDependencyMacro)
+find_dependency(quickcpplib)
```
Comment (ned14) [inline CMakeLists.txt:481]: Isn't `find_quickcpplib_library()` responsible for calling `find_package()`/`find_dependency()`? https://github.com/ned14/quickcpplib/blob/master/cmakelib/QuickCppLibUtils.cmake#L304

**Comment by ned14** [comment]
> @ned14 the installability action seems to checkout the current develop branch:
> 
> https://github.com/ned14/llfio/blob/8861d608734d52bb0a63b7fc5aae9fc17b36d634/.github/workflows/installability.yml#L41
> 
> 
> ... working as intended?

Yeah this is one of those things I keep meaning to get around to fixing. Basically, if master and develop diverge too much, the installability test fails.

> It also seems that the install componentization broke the current llfio master & develop installs in combination with the most recent quickcpplib commits due to the additional `ntkernel-error-category` target installs. This needs to be fixed by either merging this PR or reverting the commits on quickcpplib 😒

Might as well make everything consistent.

File: CMakeLists.txt (hunk 1 above)
Comment (BurningEnlightenment) [reply CMakeLists.txt:481]: This code gets injected into `llfioConfig.cmake` which cannot depend on `cmakelib` being loaded/used in the consumer project. If `find_quickcpplib_library()` behaviour is desired by the consumer it is on him to explicitly call `find_quickcpplib_library()` from his `CMakeLists.txt`. Is this problematic?

Comment (ned14) [reply CMakeLists.txt:481]: Let's get things working for you first, then cpp-pm, then I'll figure out solutions to any other breakage which keeps the previous two happy.

**Comment by BurningEnlightenment** [comment]
> Yeah this is one of those things I keep meaning to get around to fixing. Basically, if master and develop diverge too much, the installability test fails.

My original question was more about the intention: Should it test the installability of the current PR state (and master/develop branches) or should it only test the installability of the current master/develop branches i.e. the PR trigger ought to be removed.

I think the former case might be tricky to support due to PR branches not necessarily living inside this repository.

**Comment by ned14** [comment]
Ideally it should check that the PR is installable, but I think the blocker there is the helper python scripts don't currently have the ability to check anything but master and develop branch. In theory, if they could be taught to fetch more from the git repo and/or checkout explicit SHAs, that would work.

**Pull request merged**